internal/npm_install/npm_install.bzl

```diff
@@ -836,20 +834,20 @@ def _yarn_install_impl(repository_ctx):
         else:
             fail("--frozen-lockfile should not be used with yarn 2+. Just pass arguments like --immutable.")
 
-    if not repository_ctx.attr.use_global_yarn_cache:
-        yarn_args.extend(["--cache-folder", str(repository_ctx.path("_yarn_cache"))])
-    elif yarn_version == "classic":
-        # Multiple yarn rules cannot run simultaneously using a shared cache.
-        # See https://github.com/yarnpkg/yarn/issues/683
-        # The --mutex option ensures only one yarn runs at a time, see
-        # https://yarnpkg.com/en/docs/cli#toc-concurrency-and-mutex
-        # The shared cache is not necessarily hermetic, but we need to cache downloaded
-        # artifacts somewhere, so we rely on yarn to be correct.
-        yarn_args.extend(["--mutex", "network"])
+    if repository_ctx.attr.use_global_yarn_cache:
+        # According to maintainers, the yarn cache is now atomic in berry, so this flag is only needed in classic.
+        # See https://github.com/bazelbuild/rules_nodejs/pull/3195#discussion_r780719932
+        if yarn_version == "classic":
+            # Multiple yarn rules cannot run simultaneously using a shared cache.
+            # See https://github.com/yarnpkg/yarn/issues/683
+            # The --mutex option ensures only one yarn runs at a time, see
+            # https://yarnpkg.com/en/docs/cli#toc-concurrency-and-mutex
+            # The shared cache is not necessarily hermetic, but we need to cache downloaded
+            # artifacts somewhere, so we rely on yarn to be correct.
+            yarn_args.extend(["--mutex", "network"])
     else:
-        # Can't tell from documentation if Yarn Berry has any replacement for the --mutex
-        # flag. We'll have to assume it's safe to run concurrently.
-        pass
+        yarn_args.extend(["--cache-folder", str(repository_ctx.path("_yarn_cache"))])
```

Comment:
This unfortunately will add `--cache-folder` for folks who use Yarn Berry and set user_global_yarn_cache to False.

The flag DOES exist within Yarn Berry, but the behavior is to exit with a deprecation error which is not useful:

https://sourcegraph.com/github.com/yarnpkg/berry/-/blob/packages/plugin-essentials/sources/commands/install.ts?L207-219
https://sourcegraph.com/github.com/yarnpkg/berry@ce1e0c31781e3fc791cfa86a01348c4f0dba9aec/-/blob/packages/gatsby/content/advanced/error-codes.md?L295-301

I suggest ensure that refactor it like this:

```
 if yarn_version == "classic":
    if repository_ctx.attr.use_global_yarn_cache:
        ...
    else:
        ...

if yarn_version == "berry" and not repository_ctx.attr.use_global_yarn_cache:
    # Using Yarn Berry and non-default option for global_yarn_cache
    fail("Option global_yarn_cache has no effect while using Yarn Berry. Please configure it through a .yarnrc file through 'data' attribute.")
```

Reply:
yeah my intent here was the minimal change. thanks for looking up the yarn sources, if that flag just exits that way then I agree we shouldn't call it